torch_dicom/datasets/dicom.py

```diff
@@ -382,18 +383,31 @@ def load_raw_example(
         # Ensure that the path is set to DUMMY_PATH
         rec = replace(rec, path=DUMMY_PATH)
 
-        # Copy the dicom object to avoid modifying the original and remove the pixel data.
-        # We first do a shallow copy to remove pixel data followed by a deep copy.
-        dcm = dcm.copy()
-        delattr(dcm, "PixelData")
-        delattr(dcm, "_pixel_array")
-        dcm = deepcopy(dcm)
+        # We want to copy dcm before removing the pixel attributes so that we don't modify the original.
```

Comment:
The way you've commented this code is very helpful. Maybe we should consider wrapping this in a separate function as well? E.g., `def copy_dicom_without_pixels(dcm: Dataset) -> Dataset:`

Reply:
Good idea